src/main/java/seedu/internbuddy/model/company/Company.java

```diff
@@ -67,6 +72,14 @@ public Set<Tag> getTags() {
         return Collections.unmodifiableSet(tags);
     }
 
+    /**
+     * Adds a new application to the company
+     */
+    public void addApplication(Application application) {
+        this.status = status.applied();
+        this.applications.add(application);
+    }
```

Comment:
I think this should return a copy of the original application with the new application appended and status changed. (see `EditCommand.EditCompanyDescriptor` for reference)